react/tic-tac-toe/src/app/screens/Game/index.js

```diff
@@ -1,20 +1,108 @@
-import React from 'react';
+import React, { Component } from 'react';
 
 import styles from './styles.module.scss';
+
 import Board from './components/Board';
 
-function Game () {
-  return (
+class Game extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      history: [{
+        squares: Array(9).fill(null),
+      }],
+      stepNumber:0,
+      xIsNext: true,
+    };
+  }
+
+  handleClick(i) {
+        const history = this.state.history.slice(0, this.state.stepNumber + 1);
+    const current = history[history.length - 1];
+    const squares = current.squares.slice();
+    if (calculateWinner(squares) || squares[i]) {
+      return;
+    }
+    squares[i] = this.state.xIsNext ? 'X' : 'O';
+    this.setState({
+      history: history.concat([{
+        squares: squares,
+      }]),
+      stepNumber: history.length,
+      xIsNext: !this.state.xIsNext,
+    });
+  }
+
+  jumpTo(step) {
+    this.setState({
+      stepNumber: step,
+      xIsNext: (step % 2) === 0,
+    })
+  }
+
+  render() {
+    const history = this.state.history;
+    const current = history[this.state.stepNumber];
+    const winner = calculateWinner(current.squares);
+
+    const moves = history.map((step, move) => {
+      const desc = move ?
+        'Go to move #' + move :
+        'Go to game start';
+      return (
+        <li key={move}>
+          <button onClick={() => this.jumpTo(move)}>{desc}</button>
+        </li>
+      );
+    });
+
+    let status = decideGameStatus(winner,this.state.xIsNext);
+    return (
     <div className={styles.game}>
       <div className={styles.gameBoard}>
-        <Board />
+        <Board
+          squares={current.squares}
+          onClick={(i) => this.handleClick(i)}
+         />
       </div>
       <div className={styles.gameInfo}>
-        <div>{/* status */}</div>
-        <ol>{/* TODO */}</ol>
+        <div>{status}</div>
+        <ol>{moves}</ol>
       </div>
     </div>
   );
+  }
+}
+
+function decideGameStatus(winner, xIsNext) {
```

Comment:
As a convention, we usually move functions that aren't part of component logic and serves as utils to a `utils.js` file. Same for `calculateWinner`.